linguistics/src/main/java/com/yahoo/language/significance/impl/DefaultSignificanceModelRegistry.java

```diff
@@ -23,24 +24,45 @@
 public class DefaultSignificanceModelRegistry implements SignificanceModelRegistry {
 
     private final Map<Language, SignificanceModel> models;
+
     @Inject
-    public DefaultSignificanceModelRegistry(SignificanceConfig cfg) { this(new Builder(cfg)); }
-    private DefaultSignificanceModelRegistry(Builder b) {
+    public DefaultSignificanceModelRegistry(SignificanceConfig cfg) {
         this.models = new EnumMap<>(Language.class);
-        b.models.forEach((language, path) -> {
-            models.put(language,
-                    uncheck(() -> new DefaultSignificanceModel(path)));
-        });
+        ObjectMapper objectMapper = new ObjectMapper();
+        for (var model : cfg.model()) {
+            try {
+                SignificanceModelFile file = objectMapper.readValue(model.path().toFile(), SignificanceModelFile.class);
+
+                for (var pair : file.languages().entrySet()) {
+                    addModel(
+                            Language.fromLanguageTag(pair.getKey()), new DefaultSignificanceModel(pair.getValue(), file.id()));
+                }
+            } catch (IOException e) {
+                throw new RuntimeException("Failed to load model from " + model.path(), e);
+            }
+        }
     }
 
-    public DefaultSignificanceModelRegistry(HashMap<Language, Path> map) {
+    public DefaultSignificanceModelRegistry(List<Path> models) {
         this.models = new EnumMap<>(Language.class);
-        map.forEach((language, path) -> {
-            models.put(language,
-                    uncheck(() -> new DefaultSignificanceModel(path)));
-        });
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        for (var model : models) {
+            try {
+                SignificanceModelFile file = objectMapper.readValue(model.toFile(), SignificanceModelFile.class);
+                for (var pair : file.languages().entrySet()) {
+                    addModel(
+                            Language.fromLanguageTag(pair.getKey()), new DefaultSignificanceModel(pair.getValue(), file.id()));
+                }
+            } catch (IOException e) {
+                throw new RuntimeException("Failed to load model from " + model, e);
+            }
```

Comment:
This code snippet is the same as in the other constructor and can be moved to a separate method to reduce code duplication.